src/Entity/DatasetSubmission.php

```diff
@@ -1612,29 +1598,17 @@ public function getDatasetFileUri() : ?string
         return $this->datasetFileUri;
     }
 
-    /**
-     * Set the dataset file transfer status.
-     *
-     * @param string|null $datasetFileTransferStatus The dataset file transfer status.
-     *
-     * @see TRANSFER_STATUSES class constant for valid values.
-     *
-     * @return void
-     */
-    public function setDatasetFileTransferStatus(?string $datasetFileTransferStatus)
-    {
-        $this->datasetFileTransferStatus = $datasetFileTransferStatus;
-        $this->updateAvailabilityStatus();
-    }
-
     /**
      * Get the dataset file transfer status.
      *
      * @return string|null
      */
     public function getDatasetFileTransferStatus() : ?string
     {
-        return $this->datasetFileTransferStatus;
+        if ($this->fileset instanceof Fileset and $this->fileset->isDone()) {
```

Comment:
We need to figure out a way for remotely hosted datasets, because updateAvailabilityStatus() expects transfer status to be TRANSFER_STATUS_REMOTELY_HOSTED

Reply:
I think the best thing to do is just to make a whole new property called "Remotely Hosted".

Reply:
Sure, we could do that. But we need to discuss the current dataset workflow, because the availability status is controlled by this property